### PR TITLE
{NewRelic} Generate commands with codegen v2

### PR DESCRIPTION
**Pull request by ZengTaoxu**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
NewRelic

**Comment by kairu-ms** [comment]
Why it has two extensions in your PR?

**Comment by ZengTaoxu** [comment]
> Why it has two extensions in your PR?
It may have been written incorrectly when exporting. It has been deleted.

File: src/new-relic/setup.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#
+# Code generated by aaz-dev-tools
+# --------------------------------------------------------------------------------------------
+
+from codecs import open
+from setuptools import setup, find_packages
+
+
+# HISTORY.rst entry.
+VERSION = '0.1.0'
```
Comment (jsntcy) [inline src/new-relic/setup.py:13]:
```suggestion
VERSION = '1.0.0b1'
```

File: src/new-relic/HISTORY.rst [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.. :changelog:
+
+Release History
+===============
+
+0.1.0
```
Comment (jsntcy) [inline src/new-relic/HISTORY.rst:6]:
```suggestion
1.0.0b1
```

File: src/new-relic/azext_new_relic/aaz/latest/new_relic/monitor/_create.py [hunk 3]
```diff
@@ -0,0 +1,627 @@
...
+
+
+@register_command(
+    "new-relic monitor create",
+)
+class Create(AAZCommand):
+    """Create a NewRelicMonitorResource
+
+    :example: Create a NewRelicMonitorResource.
+        az new-relic monitor create --resource-group MyResourceGroup --name MyNewRelicMonitor --location eastus2euap --user-info {"first-name":"vdftzcggiref","last-name":"bcsztgqovdlmzf","email-address":"UserEmail@123.com","phone-number":"123456"} --plan-data {"billing-cycle":"MONTHLY","effective-date":"\'2022-10-25T15:14:33+02:00\'","plan-details":"newrelic-pay-as-you-go-free-live@TIDgmz7xq9ge3py@PUBIDnewrelicinc1635200720692.newrelic_liftr_payg","usage-type":"PAYG"} --account-creation-source "LIFTR" --org-creation-source "LIFTR" --tags {"key6976":"oaxfhf"}
```
Comment (kairu-ms) [inline src/new-relic/azext_new_relic/aaz/latest/new_relic/monitor/_create.py:21]: Please use shorthand syntax in the examples

File: src/new-relic/azext_new_relic/aaz/latest/new_relic/monitor/tag_rule/_create.py [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+
+
+@register_command(
+    "new-relic monitor tag-rule create",
+)
+class Create(AAZCommand):
+    """Create a TagRule
+
+    :example: Create a TagRule.
+        az new-relic monitor tag-rule create --resource-group MyResourceGroup --monitor-name MyNewRelicMonitor --name default --log-rules {"send-aad-logs":"Enabled","send-subscription-logs":"Enabled","send-activity-logs":"Enabled","filtering-tags":[]} --metric-rules {"user-email":"UserEmail@123.com","filtering-tags":[{"name":"Environment","value":"Prod","action":"Include"}]}
```
Comment (kairu-ms) [inline src/new-relic/azext_new_relic/aaz/latest/new_relic/monitor/tag_rule/_create.py:21]: Please use shorthand syntax

File: src/new-relic/azext_new_relic/aaz/latest/new_relic/monitor/tag_rule/_update.py [hunk 5]
```diff
@@ -0,0 +1,415 @@
...
+
+
+@register_command(
+    "new-relic monitor tag-rule update",
+)
+class Update(AAZCommand):
+    """Update a TagRule
+
+    :example: Update a TagRule.
+        az new-relic monitor tag-rule update --resource-group MyResourceGroup --monitor-name MyNewRelicMonitor --name default --log-rules {"send-aad-logs":"Enabled","send-subscription-logs":"Enabled","send-activity-logs":"Disabled","filtering-tags":[]}
```
Comment (kairu-ms) [inline src/new-relic/azext_new_relic/aaz/latest/new_relic/monitor/tag_rule/_update.py:21]: Please use shorthand syntax

File: src/new-relic/azext_new_relic/tests/latest/test_new_relic.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+
+class NewRelicScenario(ScenarioTest):
+
+    @AllowLargeResponse(size_kb=10240)
+    @ResourceGroupPreparer(name_prefix='cli_test_new_relic_monitor')
+    def test_new_relic_monitor(self, resource_group):
+        self.kwargs.update({
+            'new_relic_monitor_name': 'test-new-relic-monitor',
+            'user_email': 'v-taoxuzeng@microsoft.com',
```
Comment (kairu-ms) [inline src/new-relic/azext_new_relic/tests/latest/test_new_relic.py:19]: You should consider use a fake email if the fake email can work.

Comment (ZengTaoxu) [reply src/new-relic/azext_new_relic/tests/latest/test_new_relic.py:19]: Fake email cannot create success.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**